script.js

```diff
@@ -170,3 +170,31 @@ form.addEventListener('submit', (e) => {
   validateInputs();
 });
 // #endregion
+// Get All input fields and text area in the doc
+const inputFields = document.querySelectorAll('input, textarea');
+// Loop through each input field and add an "input" event listener
+inputFields.forEach((input) => {
+  input.addEventListener('input', (event) => {
+    // Get the updated value of the input field
+    const updatedValue = event.target.value;
+    // Get the key name for the local storage item (use the input field's name attribute)
+    const keyName = event.target.name;
+    // Save the updated value to local storage
+    localStorage.setItem(keyName, updatedValue);
```

Comment:
- [ ] Kindly make sure that you use the` localStorage.getItem` and `localStorage.setItem` functions to save and retrieve the form data object to the local storage.In order to be saved to the local storage the data object is turned into a string using **JSON.stringify** or similar methods. Ie: `localStorage.setItem('contactFormInput', JSON.stringify(formData)`);
- [ ] In order to be used after retrived from the local storage the saved string is turned back into an object using JSON.parse or similar methods.
```
let savedFormData = localStorage.getItem('contactFormInput');
savedFormData = JSON.parse(savedFormData);
```